docs/docs/migrate-from-v1/README.md

```diff
@@ -0,0 +1,259 @@
+
+### Editing vs Inspecting
+
+When you first open a deployed workflow, you're presented with the **inspector** version of the workflow. In this view you can see logs of past events, and select them to see the results of each step in the workflow.
+
+To edit the workflow, click the **Edit** button in the top right hand corner. This will close the inspector and allow you to edit your workflow without the distraction of logs from the production flow.
+
+### Testing Changes
+
+In the v1 workflow builder to test changes to individual steps required saving then deploying each change to run a test.
+
+We've improved this flow. You can test your changes with a new **Test** button.
+
+In addition to testing single steps, you can now selectively test portions of your workflow in isolation.
+
+<img src="./images/test-workflow-portions.png" alt="Selectively pick testing your workflow above or below the current step is now available.">
+
+Now you can also choose to limit your test runs of your workflow before _or_ after the current step.
+
+### Deploying Changes
+
+After you're happy with your changes, **deploy** them to your production workflow. Just click the **Deploy** button in the top right hand corner of the screen.
+
+After deploying your changes, your workflow is now live.
+
+## Node.js Code Step Changes
+
+There are a few changes to the Node.js code steps that you should know about. Some functions have been renamed for more clarity, and we've aligned the Node.js code steps closer to the [Component API](/docs/components/).
+
+### Code Scaffolding Format
+
+In v1, the Node.js steps would automatically scaffold new Node.js steps in this format:
+
+```javascript
+async (event, steps) {
+  // your code could be entered in here
+}
+```
+
+In v2, the new scaffolding is wrapped with a new `defineComponent` function:
+
+```javascript
+defineComponent({
+  async run({ steps, $ }) {
+    // your code can be entered here
+  }
+});
+```
+
+1. The `event` from the trigger step is still available, but within `steps.trigger.event` instead.
+2. The `$` variable has been passed into the `run` function where your code is executed.
+
+### using 3rd party packages
+
+In v1, you had to define your imports of 3rd party packages within the scaffolded function:
+
+```javascript
+async (event, steps) {
+  const axios = require('axios');
+  // your code could be entered in here
+}
+```
+
+Now, in v2 workflows you can `import` your packages in the top of the step, just like a normal Node.js module:
+
+```javascript
+import axios from 'axios';
+
+defineComponent({
+  async run({ steps, $ }) {
+    // your code can be entered here
+  }
+});
+```
+
+Allowing all of the scaffolding to be edited opens up the ability to [pass props](code/nodejs/#passing-props-to-code-steps) into your Node.js code steps, which we'll cover later.
```

Comment:
I would recommend moving this to the section where you first reference `defineComponent`. That way you can talk about why we provide the component API and what the benefits are right when the user first learns what it is.

Reply:
I really think passing dynamic props is a bit more advanced than sending HTTP requests.

I updated the sentence to hit that we'll talk about passing props with a link, but I don't want to dive into the more advanced topics right off the bat.

> These changes unify workflow development to the [Component API](/docs/components/) used by pre-built actions and also allows the [defining of props](#defining-props) from within your code steps.

Reply:
nice that works for me